### ISSUE TITLE
Issues with cert-manager 0.5.2 and its upgrade to 0.6.x

### ISSUE THREAD

**Issue by jjo** [bitnami/kube-prod-runtime#416]
FYI not currently using BKPR provided `cert-manager`, although
BKPR may face similar issues as per below details, specially
for incoming BKPR 1.2 release.

Setup: issuing TLS certs for FQDNs from CNAME wildcard pointing to AWS ELBs,
cert-manager via ACME DNS01

1. `cert-manager 0.5.2` may not support TLS certs for FQDNs that point to the actual LB by CNAME: 
On an AWS/kops:
* issue: hit by jetstack/cert-manager#1035, while there's no current 0.5 release with jetstack/cert-manager#1136 fix
* solution: use 0.6.x

2. `cert-manager 0.6.x` may require extra CRDs if upgrading from previous versions
* issue: hit by jetstack/cert-manager#1134, cert-manager showing errors at log as described in the bug
* solution: see https://github.com/jetstack/cert-manager/issues/1134#issuecomment-444210493, the helm chart has some provisions to fix this, but if deploying from manifests you'll need to do the equivalent of:
```
kubectl apply -f https://raw.githubusercontent.com/jetstack/cert-manager/v0.6.0/deploy/manifests/00-crds.yaml
```
We then fixed it by adding below `0.6` CRDs to our `cert-manager.jsonnet` manifest as:
```
// CRDs added on cluster-manager 0.6: Order, Challenge
orderCRD: kube.CustomResourceDefinition("certmanager.k8s.io", "v1alpha1", "Order"),
challengeCRD: kube.CustomResourceDefinition("certmanager.k8s.io", "v1alpha1", "Challenge"),
```

**Comment by sameersbn** [comment]
Thanks for the information @jjo.
We're working on the cert-manager upgrade and the information you've shared here is extremely helpful.

**Comment by arapulido** [comment]
Thanks @jjo !

**Comment by anguslees** [comment]
If I understand (1) correctly, we'll avoid that because we're still doing HTTP challenges.  This means we have to upgrade to cert-manager 0.6 before we switch to DNS01 challenges (which is exactly what will happen, just by luck).

For (2), it looks like these extra CRDs _were_ included in #398, so I think we're good there.

Unless we decide to cherry pick #398 to old releases (or some similar out-of-sequence git reshuffle), I don't think there are any remaining action items/warnings from this bug.  Please reopen if I missed something.
